src/ExcelDataReader/Core/BinaryFormat/XlsBiffStream.cs

```diff
@@ -126,12 +126,8 @@ public XlsBiffRecord Read()
 
             if (record != null)
             {
-                // Set readOption to loose to not cause exception here (sql reporting services)
-                if (_workbook.ReadOption == ReadOption.Strict)
-                {
-                    if (record.Bytes.Length < Position + record.Size)
-                        throw new ArgumentException(Errors.ErrorBiffBufferSize);
-                }
+                if (record.Bytes.Length < Position + record.Size)
+                    throw new ArgumentException(Errors.ErrorBiffBufferSize);
```

Comment:
It might make sense to remove this as well so that we just stop as we do if position + record "header" is past the end of the stream. Any opinions?